### DISCUSSION TITLE
About the setting of "message.shape[0]"

### DISCUSSION THREAD

**Discussion by Asecretboy** [CederGroupHub/chgnet#143] [Q&A] [answered]
I noticed in the “AtomConv layer” that "messages.shape[0]" is twice the number of bonds (where the bond count refers to the number of 'undirected edges'). I understand you're using the count of directed edges as the dimension here, but I'm curious, it whether be feasible using the count of undirected edges instead? And why did you not choose "undirected
 edge"?

**Comment by bowen-bd** [comment, marked as answer]
the message passing is directional in atom graph, so the adjacency list is essentially in the dimention of directed edges.
If you change the message passing to be undirectional, it will be dimension of undirected edge (half of directed edge).

**Reply by Asecretboy** [reply]
But in this paper, you choose the dimension of directed edges. Is it because choosing the dimension of undirected edge has too low accuracy?
